engines/e_devcrypto.c

```diff
@@ -210,7 +220,14 @@ static int cipher_init(EVP_CIPHER_CTX *ctx, const unsigned char *key,
     cipher_ctx->op = enc ? COP_ENCRYPT : COP_DECRYPT;
     cipher_ctx->mode = cipher_d->flags & EVP_CIPH_MODE;
     cipher_ctx->blocksize = cipher_d->blocksize;
+#ifdef CIOCGSESSION2
+    cipher_ctx->sess.crid = (use_softdrivers == DEVCRYPTO_USE_SOFTWARE) ?
+        CRYPTO_FLAG_SOFTWARE | CRYPTO_FLAG_HARDWARE :
+        CRYPTO_FLAG_HARDWARE;
+    if (ioctl(cfd, CIOCGSESSION2, &cipher_ctx->sess) < 0) {
+#else
     if (ioctl(cfd, CIOCGSESSION, &cipher_ctx->sess) < 0) {
+#endif
```

Comment:
control flow in preprocessor conditionals is always a bit dicey; I think I'd be happier to have an `int ret; ret = ioctl(...); if (ret < 0) {` type of flow.

Reply:
Ok